### PR TITLE
tstest/integration/vms: t.Log for VM output

### PR DESCRIPTION
**Pull request by Xe**
Previously we spewed a lot of output to stdout and stderr, even when
`-v` wasn't set. This is sub-optimal for various reasons. This patch
makes a writer out of testing.TB so that output can be shunted to
test logs.

Updates #1988

Signed-off-by: Christine Dodrill <xe@tailscale.com>

### CODE REVIEW

**Comment by Xe** [comment]
```
=== CONT  TestVMIntegrationEndToEnd/join-net/23100
    vms_test.go:539: Match for RE: "(\\$|\\>)" found: ["$" "$"] Buffer:
               __|  __|_  )
               _|  (     /   Amazon Linux 2 AMI
              ___|\___|___|

        https://aws.amazon.com/amazon-linux-2/
        8 package(s) needed for security, out of 17 available
        Run "sudo yum update" to apply all updates.
        -sh-4.2$
    vms_test.go:444: got shell for 23100
    vms_test.go:539: Sent: "sudo tailscale up --login-server http://192.168.2.14:43859\n"
    vms_test.go:539: Match for RE: "Success." found: ["Success."] Buffer: Success.
```

File: tstest/integration/vms/vms_test.go [hunk 1]
```diff
@@ -530,3 +530,17 @@ runcmd:
  - [ systemctl, enable, --now, tailscaled.service ]
  - [ curl, "{{.HostURL}}/myip/{{.Port}}", "-H", "User-Agent: {{.Hostname}}" ]
 `
+
+type testLogWriter struct {
```
Comment (bradfitz) [inline tstest/integration/vms/vms_test.go:534]: use tstest.FixLogs(t) instead: https://github.com/tailscale/tailscale/blob/v1.8.6/tstest/log.go#L18

Comment (bradfitz) [reply tstest/integration/vms/vms_test.go:534]: Oh, I guess you'll need to export (and document) some stuff instead.

**Review by josharian — COMMENTED** [review]
The main downside to this approach is that there is no guarantee that writes made to an io.Writer are line-by-line.

I'd suggest a different approach: Replace cmd.Run() with cmd.CombinedOutput(), and then print the output before the t.Fatal, if there is an error. (Or print it using t.Log even if there is no error, if you prefer.)

File: tstest/integration/vms/vms_test.go (hunk 1 above)
Comment (Xe) [reply tstest/integration/vms/vms_test.go]: I can just yoink log.Output!

**Review by bradfitz — APPROVED** [review]

**Pull request merged**